### DISCUSSION TITLE
file-locks flag not recognized as an option

### DISCUSSION THREAD

**Discussion by Barlojohn** [containers/podman#16531] [General]
I 'm trying to checkpoint a container with --file-locks and the message is :
checkpoint: unrecognized option '--file-locks'

command : podman container checkpoint  --file-locks --tcp-established <container_name>

Is this a known issue?
Podman version : 4.1.1
criu version : Version: 3.17.1

This is also reproducible with the latest podman version : v4.3.1

**Comment by mheon** [comment]
Potentially an issue with a too-old CRI-U package? @adrianreber PTAL

**Comment by adrianreber** [comment]
What version of crun/runc is being used. That would be my first suspicion.

**Reply by Barlojohn** [reply]
runc --version
runc version 1.0.0-rc95
commit: b9ee9c6314599f1b4a7f497e1f1f856fe433d3b7
spec: 1.0.2-dev
go: go1.13.15
libseccomp: 2.5.1

crun --version
crun version UNKNOWN     crun is already the newest version (100:1.2-2)
commit: ea1fe3938eefa14eb707f1d22adff4db670645d6
spec: 1.0.0

criu --version
Version: 3.17.1

Podman : Version:   3.4.2  (it is also reproducible at latest version 4.3.1)
System : Ubuntu 20.04.5 LTS"

I don't understand how it that possible. I tried a lot not to reproduce this scenario. But everytime the flag --file-locks is unknown.
And every other flag from this page : https://docs.podman.io/en/latest/markdown/podman-container-checkpoint.1.html
is readable.

Any suggestions?
Kind Regards everyone

**Reply by adrianreber** [reply]
Assuming Podman uses crun then your version of crun is too old. You need at least 1.4.0:
```shell
# git tag --contains 1575f2fed4ef3432b05a8d18b05503cf13070f36
1.4
1.4.1
1.4.2
1.4.3
1.4.4
1.4.5
1.5
1.6
1.7
1.7.1
```

Either try to update crun or switch to runc.
